src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1602,8 +1602,12 @@ Namespace Microsoft.CodeAnalysis.Operations
             Return New AnonymousObjectCreationOperation(initializers, _semanticModel, syntax, type, isImplicit)
         End Function
 
-        Private Function CreateBoundAnonymousTypePropertyAccessOperation(boundAnonymousTypePropertyAccess As BoundAnonymousTypePropertyAccess) As IPropertyReferenceOperation
+        Private Function CreateBoundAnonymousTypePropertyAccessOperation(boundAnonymousTypePropertyAccess As BoundAnonymousTypePropertyAccess) As IOperation
             Dim [property] As IPropertySymbol = DirectCast(boundAnonymousTypePropertyAccess.ExpressionSymbol, IPropertySymbol)
+            Return CreateBoundAnonymousTypePropertyAccessOperation(boundAnonymousTypePropertyAccess, [property])
```

Comment:
>Return CreateBoundAnonymousTypePropertyAccessOperation(boundAnonymousTypePropertyAccess, [property]) [](http://example.com/codeflow?start=12&length=100)

Alternatively, consider creating ```InvalidOperation``` when ```[property] Is Nothing``` and reverting changes in ```GetAnonymousTypeCreationInitializers```.

Reply:
I tried that initially. However, `TestOperationVisitor.VisitAnonymousObjectCreation()` expects the initializer target to be an `IPropertyReferenceOperation`, and I assumed there might be consumers other than test code that also expect the same.

Reply:
I think we should simply change the assertion in ```TestOperationVisitor.VisitAnonymousObjectCreation```, what it is doing is mostly driven by implementation. When we are fixing bugs, we are changing shape of ```IOperation``` tree even for success scenarios, not to mention error scenarios. The binder says, it doesn't know what the property is (that is what the bound node returns) and I think we should stick with that rather than trying to re-guess what that property might be. Otherwise, we probably should fix what the bound node returns, so that ```SemanticModel``` and ```IOperation``` tree produce consistent result.

Reply:
Thanks. Based on our offline discussion, I've left this as is, and added a check of the SemanticModel to the tests.